upup/pkg/fi/cloudup/gce/utils.go

```diff
@@ -81,7 +84,22 @@ func SafeClusterName(clusterName string) string {
 // SafeObjectName returns the object name and cluster name escaped for GCE
 func SafeObjectName(name string, clusterName string) string {
 	gceName := name + "-" + clusterName
-
+	if len(gceName) > 64 {
```

Comment:
Can I suggest we pull this function into a separate PR?  The rest of the PR has just minor quibbles and I think is basically ready to go...

Here I guess there are some conflicting goals, in that we want to preserve the suffix as it's the primary mechanism we have for recognizing the cluster, but we want to preserve the prefix because it has more signal to the user than the common sufix.  On AWS when we're limited in length we focus on the prefix (check out GetResourceName32 and GetClusterName40).

We can also tweak the deletion logic to look for the hash suffix, which would let us prioritize the name over the suffix, but would require more code changes in deletion.